core/app/[locale]/(default)/(auth)/login/page.tsx

```diff
@@ -1,55 +1,55 @@
+import { useTranslations } from 'next-intl';
 import { getTranslations, setRequestLocale } from 'next-intl/server';
 
-import { Link } from '~/components/link';
-import { Button } from '~/components/ui/button';
+import { ButtonLink } from '@/vibes/soul/primitives/button-link';
+import { SignInSection } from '@/vibes/soul/sections/sign-in-section';
 import { locales } from '~/i18n/routing';
 
-import { LoginForm } from './_components/login-form';
+import { login } from './_actions/login';
 
 export async function generateMetadata({ params }: Props) {
-  const { locale } = await params;
+  const { locale } = params;
+  const t = await getTranslations('Login');
 
   setRequestLocale(locale);
 
-  const t = await getTranslations('Login');
-
   return {
     title: t('title'),
   };
 }
 
 interface Props {
-  params: Promise<{ locale: string }>;
+  params: { locale: (typeof locales)[number] };
 }
 
-export default async function Login({ params }: Props) {
-  const { locale } = await params;
-
+export default function Login({ params: { locale } }: Props) {
   setRequestLocale(locale);
 
-  const t = await getTranslations('Login');
+  const t = useTranslations('Login');
```

Comment:
Don't we need to use `getTranslations` instead of `useTranslations` in RSCs? Maybe I'm missing something 🤔

Reply:
Only in async components, otherwise it is fine in [shared components](https://next-intl-docs.vercel.app/docs/environments/server-client-components#shared-components).

Reply:
Nice!